src/Workspaces/CSharp/Portable/Utilities/SpeculationAnalyzer.cs

```diff
@@ -339,10 +337,14 @@ protected override bool ReplacementChangesSemanticsForNodeLanguageSpecific(Synta
 
                 return false;
             }
-            else if (currentOriginalNode.Kind() == SyntaxKind.ConditionalExpression)
+            else if (currentOriginalNode.IsKind(SyntaxKind.ReturnStatement))
             {
-                var originalExpression = (ConditionalExpressionSyntax)currentOriginalNode;
-                var newExpression = (ConditionalExpressionSyntax)currentReplacedNode;
+                return !TypesAreCompatible((currentOriginalNode as ReturnStatementSyntax).Expression, (currentReplacedNode as ReturnStatementSyntax).Expression);
```

Comment:
this seems to be the only addition i can find.  can you revert everyting else and move back to this?